src/main/java/io/debezium/connector/db2/Db2Connector.java

```diff
@@ -60,4 +68,30 @@ public void stop() {
     public ConfigDef config() {
         return Db2ConnectorConfig.configDef();
     }
+
+    @Override
+    protected void validateConnection(Map<String, ConfigValue> configValues, Configuration config) {
+        ConfigValue hostnameValue = configValues.get(RelationalDatabaseConnectorConfig.HOSTNAME.name());
+        // Try to connect to the database ...
+        Db2ConnectorConfig connectorConfig = new Db2ConnectorConfig(config);
+        try (Db2Connection connection = new Db2Connection(connectorConfig.getConfig())) {
+            try {
+                connection.connect();
+                connection.execute("SELECT version()");
+                LOGGER.info("Successfully tested connection for {} with user '{}'", connection.connectionString(), connectorConfig.username());
```

Comment:
For other connectors, we invoke `connection.username()`. Let's the same here, rendering the new method below superfluous.

While at it, could you look into declaring `connectionString()` as an abstract method on `JdbcConnection`, it seems there's such method for all sub-classes.